### PR TITLE
Accordion Block

### LINKED ISSUE #60: implement accordions block
Image
Image

We can ignore the search field in this issue. This will be handled separately

Some reference pages:
- https://www.delta.com/us/en/skymiles/how-to-earn-miles/airline-partners
- https://www.delta.com/us/en/skymiles/how-to-earn-miles/convert-partner-points
- https://www.delta.com/us/en/skymiles/how-to-earn-miles/skymiles-partners-offers

### PR DESCRIPTION
**Pull request by asthabh23**
Fix #60 

Test URLs:
- Before: https://main--delta--hlxsites.hlx.page/us/en/skymiles/how-to-earn-miles/airline-partners
- After: https://issue-60--delta--hlxsites.hlx.page/drafts/asthabharga/us/en/skymiles/how-to-earn-miles/airline-partners

### CODE REVIEW

File: us/en/skymiles/blocks/accordion/accordion.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      button.textContent = tb.textContent;
+      const p = tb.closest('p');
+      p.parentNode.insertBefore(button, p);
+      p.parentNode.removeChild(p);
+    });
+  });
+
+  const parentWrapper = document.querySelectorAll('.accordion-wrapper');
+  parentWrapper.forEach((wrapper) => {
+    if (!wrapper.querySelector('.toolbar')) {
```
Comment (ramboz) [inline us/en/skymiles/blocks/accordion/accordion.js:29]: You can change this to a guard clause to save on the indentation:
```
if (wrapper.querySelector('.toolbar')) {
  return;
}

…
```

File: us/en/skymiles/styles/styles.css [hunk 2]
```diff
@@ -21,6 +21,7 @@
   --color-delta-gray-light: #f6f6f7;
   --color-delta-red: #e51937;
   --color-delta-red-dark: #98012e;
+  --color-delta-border: #dedfe1;
```
Comment (ramboz) [inline us/en/skymiles/styles/styles.css:24]: You might be able to re-use `--color-delta-gray-middle` as it should be visually close

File: us/en/skymiles/styles/styles.css [hunk 3]
```diff
@@ -481,6 +482,10 @@ main img {
   main .section {
     max-width: 375px;
   }
+
+  main .section.accordion-container {
+    width: 100%;
```
Comment (ramboz) [inline us/en/skymiles/styles/styles.css:487]: Not sure this would have any effect since sections should already be displayed as block by default, and hence have 100% width

File: us/en/skymiles/scripts/scripts.js [hunk 4]
```diff
@@ -84,14 +84,14 @@ function decorateInlineToggles(container) {
...
 
-function decorateScreenReaderOnly(container) {
-  [...container.querySelectorAll('del')].forEach((el) => {
-    const span = document.createElement('span');
-    span.classList.add('sr-only');
-    span.innerHTML = el.innerHTML;
-    el.replaceWith(span);
-  });
-}
+// function decorateScreenReaderOnly(container) {
```
Comment (ramboz) [inline us/en/skymiles/scripts/scripts.js:87]: I fixed the linting in main for this with an eslint disable. So you can probably just ignore this and revert the changes or rebase/merge the branch

File: us/en/skymiles/blocks/accordion/accordion.css [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    top: 45%;
+}
+
+.visible {
+    display: block;
+    padding-top: 20px;
+}
+
+.bold {
+    font-weight: 1200;
```
Comment (ramboz) [inline us/en/skymiles/blocks/accordion/accordion.css:106]: Does this actually work? I thought 1000 was the max, and most fonts would max out at 700 and then not show a difference (unless they have a "black" variant which is bolder than "bold")

**Review by ramboz — APPROVED** [review]
Looks good to me! Left a few comments for minor improvements

**Review by askayastha22 — CHANGES_REQUESTED** [review]
@asthabh23 We need additional styling when an optional text is included within the image cell.
Screenshot 2023-03-09 at 12 07 35 PM
Screenshot 2023-03-09 at 12 06 32 PM

**Comment by askayastha22** [comment]
@asthabh23 🐛 **Bug Alert**

https://user-images.githubusercontent.com/108121450/224128905-34d2536d-af07-4ae2-8c39-204868558b18.mov

File: us/en/skymiles/styles/styles.css (hunk 3 above)
Comment (asthabh23) [reply us/en/skymiles/styles/styles.css]: fixed in [f1d8070](https://github.com/hlxsites/delta/pull/89/commits/f1d80700d4d0f621c6466bffc47d9c5b6e4f461c)

File: us/en/skymiles/styles/styles.css (hunk 2 above)
Comment (asthabh23) [reply us/en/skymiles/styles/styles.css]: fixed in [f1d8070](https://github.com/hlxsites/delta/pull/89/commits/f1d80700d4d0f621c6466bffc47d9c5b6e4f461c)

File: us/en/skymiles/scripts/scripts.js (hunk 4 above)
Comment (asthabh23) [reply us/en/skymiles/scripts/scripts.js]: yes, I rebased my branch, fixed in [f1d8070](https://github.com/hlxsites/delta/pull/89/commits/f1d80700d4d0f621c6466bffc47d9c5b6e4f461c)

File: us/en/skymiles/blocks/accordion/accordion.css (hunk 5 above)
Comment (asthabh23) [reply us/en/skymiles/blocks/accordion/accordion.css]: this was no longer required based on code optimisations, removed it from css file

File: us/en/skymiles/blocks/accordion/accordion.js (hunk 1 above)
Comment (asthabh23) [reply us/en/skymiles/blocks/accordion/accordion.js:29]: unable to follow this particular one

**Comment by asthabh23** [comment]
@ramboz @askayastha22 Thanks for the feedbacks
I have incorporated them in my latest commit, fixed the bug and also added styling for the optional text.
Please have a look and merge into main if all fine :)

File: us/en/skymiles/blocks/accordion/accordion.js (hunk 1 above)
Comment (ramboz) [reply us/en/skymiles/blocks/accordion/accordion.js:29]: The idea of a _guard clause_ is just to reverse the conditional and exit early. that way you avoid indenting the main code and it usually makes it easier to read and maintain as the core of the method is essentially fully contained at the end in its own block.

It's nothing blocking the PR, but just a best practice that usually helps with long-term maintainability.

If you are not familiar with that pattern, here are a few quick examples:
- https://www.codementor.io/@clintwinter/use-guard-clauses-for-cleaner-code-1rrsczgwxp
- https://www.carloscaballero.io/refactoring-guard-clauses/

**Comment by ramboz** [comment]
@askayastha22 Care to re-review now that @asthabh23 has addressed the remaining points?

**Comment by askayastha22** [comment]
@ramboz Yeah, I tested the new changes. Looks good. 👍
